
* server/main.c (main): Bail out *before* signalling the server
  if the config is bad. (as per the claim in the docs!)

https://bugzilla.redhat.com/show_bug.cgi?id=814645
http://svn.apache.org/viewvc?view=revision&revision=1328345

Upstream-Status: Backport

--- httpd-2.4.2/server/main.c.restart
+++ httpd-2.4.2/server/main.c
@@ -671,6 +671,11 @@ int main(int argc, const char * const ar
         }
     }
 
+    /* If our config failed, deal with that here. */
+    if (rv != OK) {
+        destroy_and_exit_process(process, 1);
+    }
+
     signal_server = APR_RETRIEVE_OPTIONAL_FN(ap_signal_server);
     if (signal_server) {
         int exit_status;
@@ -680,11 +685,6 @@ int main(int argc, const char * const ar
         }
     }
 
-    /* If our config failed, deal with that here. */
-    if (rv != OK) {
-        destroy_and_exit_process(process, 1);
-    }
-
     apr_pool_clear(plog);
 
     if ( ap_run_open_logs(pconf, plog, ptemp, ap_server_conf) != OK) {
